packages/cli/src/s3-store.ts

```diff
@@ -75,13 +78,33 @@ export class S3Store implements IKVStore {
   })
 
   constructor(networkName: string, bucketName: string, customEndpoint?: string) {
+    this.#bucketName = bucketName
+    this.#customEndpoint = customEndpoint
     this.#storeMap = new S3StoreMap(networkName, bucketName, customEndpoint)
   }
 
   get networkName(): string {
     return this.#storeMap.networkName
   }
 
+  async init(): Promise<void> {
+    // Check if ELP bucket is used
+    if (this.networkName === Networks.MAINNET) {
+      const s3 = new AWSSDK.S3()
+      try {
+        await s3.headBucket({ Bucket: `${this.#bucketName}/ceramic/elp` }).promise()
+        // Bucket exists and needs to be used
+        console.warn(
```

Comment:
Can we thread the DiagnosticsLogger into this class and use that instead of using the console directly?